src/main.ts

```diff
@@ -303,12 +303,18 @@ async function push(
     core.debug("Skipping push, no named modules detected");
     return skip();
   }
+  // We want push to succeed without additional user configuration even if the action is being run on an
+  // enterprise GitHub instance. Because enterprise GitHub instances can have an arbitrary URL, the Buf CLI with not be
+  // able to automatically detect the source control url, so we set it explicitly. 
+  const sourceControlUrl = `${context.serverUrl}/${context.repo.owner}/${context.repo.repo}/commit/${context.sha}`;
```

Comment:
Could have tried to make this conditional and only set this if we are on an enterprise instance, but it seems better/simpler to always set it and ensure it works everywhere instead of having multiple code paths.